src/colr.cc

```diff
@@ -0,0 +1,1069 @@
+      !subtable.ReadU24(&paintOffset) ||
+      !subtable.ReadS16(&scaleX) ||
+      (!uniform && !subtable.ReadS16(&scaleY)) ||
+      (aroundCenter && (!subtable.ReadS16(&centerX) ||
+                        !subtable.ReadS16(&centerY))) ||
+      (var && !subtable.ReadU32(&varIndexBase))) {
+    return OTS_FAILURE_MSG("Failed to read Paint[Var]Scale[...]");
+  }
+
+  if (paintOffset >= length) {
+    return OTS_FAILURE_MSG("Paint[Var]Scale[...] paint offset out of bounds");
+  }
+
+  if (!ParsePaint(font, data + paintOffset, length - paintOffset, state)) {
+    return OTS_FAILURE_MSG("Failed to parse paint for Paint[Var]Scale[...]");
+  }
+
+  return true;
+}
+
+bool ParsePaintRotate(const ots::Font* font,
+                      const uint8_t* data, size_t length,
+                      colrState& state,
+                      bool var, bool aroundCenter)
+{
+  ots::Buffer subtable(data, length);
+
+  uint32_t paintOffset;
+  F2DOT14 angle;
+  FWORD centerX, centerY;
+  VarIdxBase varIndexBase;
+
+  if (!subtable.Skip(1) ||
+      !subtable.ReadU24(&paintOffset) ||
+      !subtable.ReadS16(&angle) ||
+      (aroundCenter && (!subtable.ReadS16(&centerX) ||
+                        !subtable.ReadS16(&centerY))) ||
+      (var && !subtable.ReadU32(&varIndexBase))) {
+    return OTS_FAILURE_MSG("Failed to read Paint[Var]Rotate[...]");
+  }
+
+  if (paintOffset >= length) {
+    return OTS_FAILURE_MSG("Paint[Var]Rotate[...] paint offset out of bounds");
+  }
+
+  if (!ParsePaint(font, data + paintOffset, length - paintOffset, state)) {
+    return OTS_FAILURE_MSG("Failed to parse paint for Paint[Var]Rotate[...]");
+  }
+
+  return true;
+}
+
+bool ParsePaintSkew(const ots::Font* font,
+                    const uint8_t* data, size_t length,
+                    colrState& state,
+                    bool var, bool aroundCenter)
+{
+  ots::Buffer subtable(data, length);
+
+  uint32_t paintOffset;
+  F2DOT14 xSkewAngle, ySkewAngle;
+  FWORD centerX, centerY;
+  VarIdxBase varIndexBase;
+
+  if (!subtable.Skip(1) ||
+      !subtable.ReadU24(&paintOffset) ||
+      !subtable.ReadS16(&xSkewAngle) ||
+      !subtable.ReadS16(&ySkewAngle) ||
+      (aroundCenter && (!subtable.ReadS16(&centerX) ||
+                        !subtable.ReadS16(&centerY))) ||
+      (var && !subtable.ReadU32(&varIndexBase))) {
+    return OTS_FAILURE_MSG("Failed to read Paint[Var]Skew[...]");
+  }
+
+  if (paintOffset >= length) {
+    return OTS_FAILURE_MSG("Paint[Var]Skew[...] paint offset out of bounds");
+  }
+
+  if (!ParsePaint(font, data + paintOffset, length - paintOffset, state)) {
+    return OTS_FAILURE_MSG("Failed to parse paint for Paint[Var]Skew[...]");
+  }
+
+  return true;
+}
+
+bool ParsePaintComposite(const ots::Font* font,
+                         const uint8_t* data, size_t length,
+                         colrState& state)
+{
+  ots::Buffer subtable(data, length);
+
+  uint32_t sourcePaint;
+  uint8_t compositeMode;
+  uint32_t backdropPaint;
```

Comment:
Should be `sourcePaintOffset` and `backdropPaintOffset`.

Reply:
Done.